storybookjs/storybook
# dependencies: Update react-syntax-highlighter to fix transitive vulnerability

Issue: https://github.com/storybookjs/storybook/issues/16163 https://github.com/storybookjs/storybook/issues/16848

Vulnerability stemming from PrismJS which is a part of react-syntax-highlighter. Similar to https://github.com/storybookjs/storybook/pull/17116 but with further version updates. If that PR is updated and merged first then this one can be closed.

GitHub Vulnerability link: https://github.com/advisories/GHSA-hqhp-5p83-hx96
Fix PR reference for react-syntax-highlighter: react-syntax-highlighter/react-syntax-highlighter#430
Fix commit in react-syntax-highlighter: react-syntax-highlighter/react-syntax-highlighter@20d9444


## What I did
Upgraded the react-syntax-highlighter package in both addons/storysource && lib/components. Ran `yarn test` and there were no additional test failures.


## How to test

- [No Functionality should be impacted ] Is this testable with Jest or Chromatic screenshots?
- [ No] Does this need a new example in the kitchen sink apps?
- [ No] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.
